pkg/archive/wrap.go

```diff
@@ -22,7 +23,7 @@ import (
 //
 // FIXME: stream content instead of buffering
 // FIXME: specify permissions and other archive metadata
-func Generate(input ...string) (Archive, error) {
+func Generate(input ...string) (io.ReadCloser, error) {
```

Comment:
The only place I see it used outside of unit tests is `DetectContextFromRemoteURL` `builder/remote.go`. This indeed needs an `io.ReadCloser`, but it feels like the call to `ioutil.NopCloser` should move there, so that `Generate` is not imposing unnecessary closer semantics.